File: lab2/lab2.py
count = 0

number = int(input("Please enter a positive integer greater than 9: "))



while number > 9 :
    print(number," -> ",end="")
    digitSum = 0
    tempNum = number
    while tempNum > 0:
        digitSum += tempNum%10
        tempNum //= 10
    number = digitSum
    count += 1
print(number)

print("Final value: ", number)
print("Total steps: ", count)

number = int(input("Enter a number: "))

a = ""

while number < 10 or number > 100:
    number = int(input("Enter a number: "))

for i in range(1,number+1):
    a = ""
    if i % 7 == 0:
        continue
    if i % 3 == 0 and i % 5 == 0:
        a += "FizzBuzz"
    elif i % 3 == 0:
        a += "Fizz"
    elif i % 5 == 0:
        a += "Buzz"
    else:
        print(i)
        continue
    print(a)

number = int(input("Enter a number: "))

while number > 9 or number < 3:
    number = int(input("Enter a number: "))

for i in range(1,(number*2)):
    k = number - abs(number-i)
    for j in range(1,k+1):
        print(j,end=" ")
    print()

File: lab2/lab2.cpp
#include <iostream>

using namespace std;
int main() {
    cout << "Please enter a positive integer greater than 9: " << endl;
    int x = 0;
    cin >> x;
    int count = 0;
    while (x > 9) {
        cout << x << " -> ";
        int digitSum = 0;
        int tempNumber = x;
        while (tempNumber > 0) {
            digitSum += tempNumber % 10;
            tempNumber /= 10;
        }
        x = digitSum;
        count++;
    }
    cout << x << endl;
    cout << "Final Value : " << x << endl;
    cout << "Total Steps: " <<count << endl;


    int number = 0;
    string a = "";
    cout << "Enter a number: ";
    cin >> number;
    while (number < 10 || number > 100) {
        cout << "Enter a number: ";
        cin >> number;
    }
    for (int i = 1; i < number+1; i++) {
        a = "";
        if (i%7 == 0) {
            cout << i << " is skipped"<<endl;
            continue;
        }
        if (i%5 == 0 && i%3 == 0) {
            a = "FizzBuzz";
        }else if (i%3 == 0) {
            a = "Fizz";
        }else if (i%5 == 0) {
            a = "Buzz";
        }else {
            cout << i << endl;
            continue;
        }
        cout << a << endl;
    }

    int m = 0;
    cout<<"Enter a Number: ";
    cin >> m;
    while (m > 9 || m < 3) {
        cout<<"Enter a Number: ";
        cin >> m;
    }

    for (int i = 1; i < (m*2) ; i++) {
        int k = m - abs(m-i);
        for (int j = 1; j < k+1 ; j++) {
            cout<<j;
        }
        cout<<endl;
    }
    return 0;




}